### ISSUE TITLE
In mv command, there should be en error if destination not present.

### ISSUE THREAD

**Issue by gagan-bansal** [shelljs/shelljs#1146]
### Node version (or tell us if you're using electron or some other framework):
node version: v18.17.1
### ShellJS version (the most recent version/Github branch you see the bug on):
ShellJS version: 0.8.5
### Operating system:
Ubuntu 23.04
### Description of the bug:
In `mv` command if the destination directory does not exist, the stderr should have the error message.

### Example ShellJS command to reproduce the error:

On linux shell, there is an error message.

```sh
$ touch test-file.txt
$ mv test-file.txt /path/doesnot/exist/
mv: cannot move 'test-file.txt' to '/path/doesnot/exist/': No such file or directory
```
Javascript:
```javascript
const shell = require('shelljs')
var resp = shell.mv('./test-file.txt', '/path/doesnot/exist/')
console.log(resp.stderr)
// null
```

**Comment by nfischer** [comment]
I agree, this should ideally return an error. Some comments about this:

1. `mv test-file.txt does-not-exist/` should throw an error because `does-not-exist/` must be interpreted as a folder (due to the trialing slash). This is the case which is broken in shelljs: the behavior is actually correct (it does not copy the file), but it needs to print an error message about the issue.
2.  `mv test-directory does-not-exist/` should **not** throw an error; `test-directory` is a directory, so it will be renamed to a directory named `does-not-exist`. This case works as expected in shelljs.
3. `mv test-file.txt other-test-file.txt does-not-exist` is supposed to throw an error; if there are 3 arguments, then the last argument must be a folder which already exists. This case works as expected in shelljs.
